src/utils/src/promise.js

```diff
@@ -0,0 +1,95 @@
+import Promise from 'es6-promise';
+
+/**
+ * @private
+ */
+const noop = function() {};
+
+/**
+ * @private
+ */
+function resolveWith(value) {
+  if (value && typeof value.then === 'function') {
+    return value;
+  }
+
+  return Promise.resolve(value);
+}
+
+/**
+ * @private
+ */
+export class Queue {
```

Comment:
I have looked at other ones but I would prefer to just use this one. The amount of code is small and easy to understand.